### PR TITLE
Dataminr Pulse Release v100

### PR DESCRIPTION
**Pull request by crestdatasystems**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Dataminr Pulse's AI-powered, real-time intelligence integrates into Cortex XSOAR workflows for faster detection and response.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/DataminrPulse/Integrations/DataminrPulse/DataminrPulse.yml [hunk 1]
```diff
@@ -0,0 +1,766 @@
...
+    - contextPath: DataminrPulse.Alerts.subCaption.bullets.content
+      description: Content from information about Dataminr event derived exclusively from the attributed source.
+      type: String
+    - contextPath: DataminrPulse.Alerts.userRecentImages
+      description: User's recent images.
+      type: Unknown
+    - contextPath: DataminrPulse.Alerts.userTopHashtags
+      description: User's top hashtags.
+      type: Unknown
+  dockerimage: demisto/python3:3.10.11.57293
```
[the remark this answers is not in the document]
Comment (crestdatasystems) [reply Packs/DataminrPulse/Integrations/DataminrPulse/DataminrPulse.yml]: @thefrieddan1 updated the docker image for scripts.

File: Packs/DataminrPulse/Scripts/DataminrPulseDisplayRelatedAlerts/DataminrPulseDisplayRelatedAlerts.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            # Table headers.
+            headers = ["Alert Type", "Alert ID", "Caption", "Alert URL", "Watchlist Name", "Alert Time",
+                       "Alert Location", "Post Link", "Is source verified", "Publisher Category"]
+            human_readable = tableToMarkdown(
+                heading, hr_outputs, headers, removeNull=True, url_keys=['Post Link', 'Alert URL'])
+
+            return_results(CommandResults(readable_output=human_readable))
+
+    except Exception as e:
+        return_results(CommandResults(readable_output=f"\n#### Could not find related alerts Information. \n {e}"))
```
[the remark this answers is not in the document]
Comment (crestdatasystems) [reply Packs/DataminrPulse/Scripts/DataminrPulseDisplayRelatedAlerts/DataminrPulseDisplayRelatedAlerts.py:46]: @thefrieddan1 intentionally we're returning the return_results here as we're not able to return markdown with return_error. 
In the below Screenshots, you can see the difference,

With return_results:

return_result

With return_error:
return_error

**Comment by crestdatasystems** [comment]
@thefrieddan1, Thanks for taking time and reviewing the PR. I've replied/resolved your comments please let me know if you need anything else from our side.

**Comment by crestdatasystems** [comment]
@efelmandar, thanks for reviewing and providing your insights on this PR. Please find my answers below for your queries.

**Incident Fields:**
1.  We've checked all the commonTypes fields initially and we've used those for some of the fields however rest of the other fields are related to the content pack specifically that's why we've created new ones.
2. That's actually a good catch, considering performance impact we've revisited our field list and set `unsearchable` to `true` for 25 fields.

**Playbook**
We've extracted IOCs in the new Key instead of `ExtractIndicators` because the `ExtractIndicators` key will store all the historic indicators in the context however while running the "Retrieve Alerts For IOCs - Dataminr Pulse" playbook we just want to query the current indicators not all the past indicators that's why we created a new key and will replace the old indicators with the current ones. This way will reduce the indicator load on the Dataminr side as well.

Please refer to the screenshot where you can see in the `ExtractedIndicatorsForDataminrPulse` key we've current IOCs only whereas in `ExtractIndicators` we have past IOCs as well. Please let me know if that doesn't follow the XSOAR standards.
 

image

**Comment by efelmandar** [comment]
@crestdatasystems Thanks for the detailed response. I understand and appreciate your reasoning behind the changes. I'm approving the pull request based on your explanations - @thefrieddan1 FYI

Thank you again for your contribution, your efforts are greatly appreciated!

**Comment by efelmandar** [comment]
@crestdatasystems Please just remove all the `ignore=IF115` from the pack-ignore file as it should not fail on this validation now that you've changed the `unsearchable` to `true`

**Comment by crestdatasystems** [comment]
> @crestdatasystems Please just remove all the `ignore=IF115` from the pack-ignore file as it should not fail on this validation now that you've changed the `unsearchable` to `true`

@efelmandar We've not removed all the fields from `unsearchable` cause some of them are actually required that's why we've kept this in the ignore.

**Comment by thefrieddan1** [comment]
Hi @crestdatasystems, 
Please merge from master. 
Some of the CI tests are failing since new changes were introduced in master.
https://code.pan.run/xsoar/content/-/pipelines/5272612

Thanks

**Review by thefrieddan1 — APPROVED** [review]
Thank you for your contribution, and detailed demo attached.

**Pull request merged**